tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.Generated.cs

```diff
@@ -12,287 +12,291 @@ private static NativeCallTargetDefinition[] GetDefinitionsArray()
-                new("System.Data.SqlClient", "System.Data.SqlClient.SqlCommand", "ExecuteScalar",  new[] { "System.Object" }, 4, 0, 0, 4, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteScalarIntegration"),
-                new("System.Data.SqlClient", "System.Data.SqlClient.SqlCommand", "ExecuteScalarAsync",  new[] { "System.Threading.Tasks.Task`1<System.Object>", "System.Threading.CancellationToken" }, 4, 0, 0, 4, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteScalarAsyncIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteDbDataReader",  new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteReaderWithBehaviorIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteNonQuery",  new[] { "System.Int32" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteNonQueryIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteNonQuery",  new[] { "System.Int32", "System.Data.CommandBehavior" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteNonQueryWithBehaviorIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteReader",  new[] { "System.Data.SQLite.SQLiteDataReader" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteReaderIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteReader",  new[] { "System.Data.SQLite.SQLiteDataReader", "System.Data.CommandBehavior" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteReaderWithBehaviorIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteScalar",  new[] { "System.Object" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteScalarIntegration"),
-                new("System.Data.SQLite", "System.Data.SQLite.SQLiteCommand", "ExecuteScalar",  new[] { "System.Object", "System.Data.CommandBehavior" }, 1, 0, 0, 2, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteScalarWithBehaviorIntegration"),
+                new("Microsoft.Data.SqlClient", "Microsoft.Data.SqlClient.SqlCommand", "ExecuteDbDataReader",  new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" },  new ushort[] {  }, 1, 0, 0, 3, 65535, 65535, assemblyFullName, "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.CommandExecuteReaderWithBehaviorIntegration"),
```

Comment:
we should accept a null array, this seems like a waste of allocations.

Reply:
Yeah, that would also play into the case of where we don't need to pass any arguments to the integration [as I mentioned](https://github.com/DataDog/dd-trace-dotnet/pull/1992/files/cc0fcd90b8cc32b0299866676ca175f5622bfcd3..d6b9d1680a1ca89c8c288c46284c6c00bffc1766#diff-342b800e1ef51ce4ef957ca3672902f7b39ca419ccb9b4e2fce023647ce75a89) - an empty array would mean "no arguments", as opposed to `null` which would mean "all arguments"

Reply:
Fixed both issues in f9b0cf01e11c2487c2f6075a738c919587022ecb . `null` results in us not using the feature, and an empty array will allow us to use the feature by passing zero arguments to the instrumentation method